tests/test_create_standby_with_pgdata.py

```diff
@@ -0,0 +1,86 @@
+import pgautofailover_utils as pgautofailover
+from nose.tools import *
+
+import os
+import subprocess
+
+cluster = None
+node1 = None
+node2 = None
+node3 = None
+
+def setup_module():
+    global cluster
+    cluster = pgautofailover.Cluster()
+
+def teardown_module():
+    cluster.destroy()
+
+def test_000_create_monitor():
+    monitor = cluster.create_monitor("/tmp/sb-from-pgdata/monitor")
+    monitor.run()
+
+def test_001_init_primary():
+    global node1
+    node1 = cluster.create_datanode("/tmp/sb-from-pgdata/node1")
+    node1.create()
+    node1.run()
+    assert node1.wait_until_state(target_state="single")
+
+def test_002_create_t1():
+    node1.run_sql_query("CREATE TABLE t1(a int)")
+    node1.run_sql_query("INSERT INTO t1 VALUES (1), (2)")
+
+def test_003_init_secondary():
+    global node2
+
+    # fail the registration of a node2 by using a PGDATA directory that has
+    # already been created, with another system_identifier (initdb creates a
+    # new one each time)
+    p = subprocess.Popen(["sudo", "-E", '-u', os.getenv("USER"),
+                          'env', 'PATH=' + os.getenv("PATH"),
+                          "pg_ctl", "initdb", "-s", "-D",
+                          "/tmp/sb-from-pgdata/node2"])
+    assert(p.wait() == 0)
+
+    node2 = cluster.create_datanode("/tmp/sb-from-pgdata/node2")
+    try:
+        node2.create()
+    except Exception as e:
+        # we want to see the failure here
+        print(e)
+
+        print("Failed as expected, cleaning up")
+        print("rm -rf /tmp/sb-from-pgdata/node2")
+        p = subprocess.Popen(["sudo", "-E", '-u', os.getenv("USER"),
+                          'env', 'PATH=' + os.getenv("PATH"),
+                          "rm", "-rf", "/tmp/sb-from-pgdata/node2"])
+        assert(p.wait() == 0)
```

Comment:
Right now this test always succeeds, because of the `try` ... `except`. Something like this should be better, because then it actually fails when `node2.create()` doesn't raise an exception.
```suggestion
@raises(Exception)
def test_004_create_raises_error():
    try:
        node2.create()
    except Exception as e:
        # we want to see the failure here
        print(e)
        raise
        
def test_005_cleanup_after_failure():
        print("Failed as expected, cleaning up")
        print("rm -rf /tmp/sb-from-pgdata/node2")
        p = subprocess.Popen(["sudo", "-E", '-u', os.getenv("USER"),
                          'env', 'PATH=' + os.getenv("PATH"),
                          "rm", "-rf", "/tmp/sb-from-pgdata/node2"])
        assert(p.wait() == 0)
```

Reply:
Good point. I played around with it but failed to come to the try/except/raise idea. Thanks!